packages/compiler/core/parser.ts

```diff
@@ -964,6 +979,17 @@ export function parse(code: string | SourceFile, options: ParseOptions = {}): Ca
     return expr;
   }
 
+  function parseStandaloneReferenceExpression() {
+    const expr = parseReferenceExpression();
+    if (
+      expr.kind === SyntaxKind.TypeReference &&
```

Comment:
Isn't expr.kind alway SyntaxKind.TypeReference here

Reply:
yes that is correct, think I might have had something different at some point and didn't cleanup